recursion.rb

```diff
@@ -1,14 +1,54 @@
 
+# RC n! = n * (n-1)!
+# BC 1! = 1
+# BC 0! = 1
 def fact(n)
+  return 1 if n == 1 || n == 0
+  return n * fact(n-1)
 end
 
+# 1 1 2 3 5 8 13 ...
+# BC fib(1) => 1
+# BC fib(2) => 1
+# fib(3) => 2 => fib(2) + fib(1)
+# fib(4) => 3 => fib(3) + fib(2)
+# fib(5) => 5
+# fib(6) => 8
+# fib(n) => fib(n-1) + fib(n-2)
+# RC
 def fib(n)
+  return n if n == 1 || n == 0
+  fibonacci = fib(n - 1) + fib(n - 2)
+  return fibonacci
 end
 
+# return boolean value, return strings, BC - use the length of the word in, RC - every time you test you strip off letters
+# BC s.length == 0 || s.length == 1
 def pal(s)
+  return true if s.length == 0 || s.length == 1
+  z = s.delete(" ").split("")
+  x = z.shift
+  y = z.pop
```

Comment:
You should use better variable names here for readability -- first/last or something like that.